xc/device.ml

```diff
@@ -1493,7 +1493,13 @@ module Dm = struct
 
     let qmp_event_handle domid qmp_event =
       (* This function will be extended to handle qmp events *)
-      debug "Got QMP event, domain-%d: %s" domid qmp_event.event
+      debug "Got QMP event, domain-%d: %s" domid qmp_event.event;
+      qmp_event.data >>= fun d ->
+      match d with
+      | RTC_CHANGE timeoffset ->
+        with_xs (fun xs ->
+          xs.Xs.write ("/vm/" ^ (Uuidm.to_string (Xenops_helpers.uuid_of_domid ~xs domid)) ^ "/rtc/timeoffset") (string_of_int timeoffset)
```

Comment:
From the design here ( https://code.citrite.net/projects/XSC/repos/xs-documents/browse/designs/qemu-upstream/design.txt?at=f546a53424ca8d823b55010d26978e93c5d1bf68#668 ) the offset sent by QEMU should be added to the existing timeoffset field for the VM.

Reply:
Updated, thanks.